File: 1638_count_substrings_that_differ_by_one_character/Solution.swift
// LeetCode 1638 - Count Substrings That Differ by One Character
// https://leetcode.com/problems/count-substrings-that-differ-by-one-character/

class Solution {
    func countSubstrings(_ s: String, _ t: String) -> Int {
        let S = Array(s), T = Array(t)
        var ans = 0
        for i in 0..<S.count {
            for j in 0..<T.count {
                var diff = 0
                var k = 0
                while k < min(S.count - i, T.count - j) {
                    if S[i + k] != T[j + k] { diff += 1 }
                    if diff == 1 { ans += 1 }
                    else if diff > 1 { break }
                    k += 1
                }
            }
        }
        return ans
    }
}

File: 1638_count_substrings_that_differ_by_one_character/Solution.java
// LeetCode 1638 - Count Substrings That Differ by One Character
// https://leetcode.com/problems/count-substrings-that-differ-by-one-character/

class Solution {
    public int countSubstrings(String s, String t) {
        int ans = 0;
        for (int i = 0; i < s.length(); i++) {
            for (int j = 0; j < t.length(); j++) {
                int diff = 0;
                for (int k = 0; k < Math.min(s.length() - i, t.length() - j); k++) {
                    if (s.charAt(i + k) != t.charAt(j + k)) diff++;
                    if (diff == 1) ans++;
                    else if (diff > 1) break;
                }
            }
        }
        return ans;
    }
}
